io/eolearn/io/processing_api.py

```diff
@@ -0,0 +1,205 @@
+''' An input task for the `sentinelhub processing api <https://docs.sentinel-hub.com/api/latest/reference/>`
+'''
+import json
+import logging
+from copy import deepcopy
+import datetime as dt
+import numpy as np
+
+from sentinelhub import WebFeatureService, MimeType, SentinelHubDownloadClient, DownloadRequest, SHConfig
+import sentinelhub.sentinelhub_request as shr
+
+from eolearn.core import EOPatch, EOTask, FeatureParser
+
+LOGGER = logging.getLogger(__name__)
+
+
+class SentinelHubProcessingInput(EOTask):
+    ''' A processing API input task that loads 16bit integer data and converts it to a 32bit float feature.
+    '''
+    def __init__(self, data_source, size=None, resolution=None, bands_feature=None, bands=None, additional_data=None,
+                 maxcc=1.0, time_difference=None, cache_folder=None, max_threads=5):
+        """
+        :param size_x: Number of pixels in x dimension.
+        :type size_x: int
+        :param size_y: Number of pixels in y dimension.
+        :type size_y: int
+        :param time_range: A range tuple of (date_from, date_to), defining a time range from which to acquire the data.
+        :type time_range: tuple(str, str)
+        :param bands_feature: Target feature into which to save the downloaded images.
+        :type bands_feature: tuple(sentinelhub.FeatureType, str)
+        :param bands: An array of band names.
+        :type bands: list[str]
+        :param additional_data: A list of additional data to be downloaded, such as SCL, SNW, dataMask, etc.
+        :type additional_data: list[tuple(sentinelhub.FeatureType, str)]
+        :param maxcc: Maximum cloud coverage.
+        :type maxcc: float
+        :param time_difference: Minimum allowed time difference, used when filtering dates, None by default.
+        :type time_difference: datetime.timedelta
+        :param cache_folder: Path to cache_folder. If set to None (default) requests will not be cached.
+        :type cache_folder: str
+        :param max_threads: Maximum threads to be used when downloading data.
+        :type max_threads: int
+        """
+        self.size = size
+        self.resolution = resolution
+        self.data_source = data_source
+        self.maxcc = maxcc
+        self.time_difference = dt.timedelta(seconds=1) if time_difference is None else time_difference
+        self.cache_folder = cache_folder
+        self.max_threads = max_threads
+
+        self.bands_feature = bands_feature
```

Comment:
It might be useful to parse this as well.

Reply:
Done.